libs/vm_monitor/src/filecache.rs

```diff
@@ -59,8 +59,15 @@ pub struct FileCacheConfig {
     spread_factor: f64,
 }
 
-impl Default for FileCacheConfig {
-    fn default() -> Self {
+impl FileCacheConfig {
+    pub fn new(in_memory: bool) -> Self {
+        match in_memory {
+            true => Self::default_in_memory(),
+            false => Self::default_on_disk(),
+        }
+    }
```

Comment:
I think it might be more clear to inline this to the caller and remove this function.

Reply:
Fair enough, done